exporter/common.go

```diff
@@ -13,62 +14,106 @@ import (
 
-		for _, collection := range collections {
+		for _, collection := range filterInCollections {
 			matchExpressions = append(matchExpressions,
 				bson.D{{Key: "name", Value: primitive.Regex{Pattern: collection, Options: "i"}}})
 		}
 
 		filter = bson.D{{Key: "$or", Value: matchExpressions}}
 	}
 
-	databases, err := client.Database(database).ListCollectionNames(ctx, filter)
+	collections, err := client.Database(database).ListCollectionNames(ctx, filter)
 	if err != nil {
 		return nil, errors.Wrap(err, "cannot get the list of collections for discovery")
 	}
 
-	return databases, nil
+	return collections, nil
 }
 
-func databases(ctx context.Context, client *mongo.Client, exclude []string) ([]string, error) {
+// databases returns the list of databases matching the filters.
+// - filterInNamespaces: Include only the database names matching the any of the regular expressions in this list.
+//                       Case will be ignored because the function will automatically add the ignore case
+//                       flag to the regular expression.
+// - exclude: List of databases to be excluded. Usefull to ignore system databases.
+func databases(ctx context.Context, client *mongo.Client, filterInNamespaces []string, exclude []string) ([]string, error) {
 	opts := &options.ListDatabasesOptions{NameOnly: pointer.ToBool(true), AuthorizedDatabases: pointer.ToBool(true)}
+
+	filter := bson.D{}
+
+	if excludeFilter := makeExcludeFilter(exclude); excludeFilter != nil {
+		filter = append(filter, *excludeFilter)
+	}
+
+	if namespacesFilter := makeDBsFilter(filterInNamespaces); namespacesFilter != nil {
+		filter = append(filter, *namespacesFilter)
+	}
+
+	dbNames, err := client.ListDatabaseNames(ctx, filter, opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot get the database names list")
+	}
+
+	return dbNames, nil
+}
+
+func makeExcludeFilter(exclude []string) *primitive.E {
 	filterExpressions := []bson.D{}
 	for _, dbname := range exclude {
 		filterExpressions = append(filterExpressions,
 			bson.D{{Key: "name", Value: bson.D{{Key: "$ne", Value: dbname}}}},
 		)
 	}
 
-	filter := bson.D{{Key: "$and", Value: filterExpressions}}
+	if len(filterExpressions) == 0 {
+		return nil
+	}
 
-	dbNames, err := client.ListDatabaseNames(ctx, filter, opts)
-	if err != nil {
-		return nil, errors.Wrap(err, "cannot get the database names list")
+	return &primitive.E{Key: "$and", Value: filterExpressions}
+}
+
+// makeDBsFilter creates a filter to list all databases or only the ones in the specified
+// namespaces list. Namespaces have the form of <db>.<collection> and the collection name
+// can have a dot. Example: db1.collection.one -> db: db1, collection: collection.one
+// db1, db2.col2, db3.col.one will produce [db1, db2, db3]
+//
+func makeDBsFilter(filterInNamespaces []string) *primitive.E {
+	dbs := []string{}
+
+	for _, namespace := range filterInNamespaces {
+		parts := strings.Split(namespace, ".")
+		if len(parts) > 0 {
```

Comment:
Is this if necessary?

Reply:
+1, even if there is no dot len(parts) is always > 0